File: src/main/java/com/mudhut/nudge/users/repositories/PhoneVerificationTokenRepository.java
package com.mudhut.nudge.users.repositories;

import org.springframework.data.jpa.repository.JpaRepository;

import com.mudhut.nudge.users.entities.PhoneVerificationToken;

import java.util.Optional;

public interface PhoneVerificationTokenRepository extends JpaRepository<PhoneVerificationToken, Long> {
    Optional<PhoneVerificationToken> findByCode(String code);
}

File: src/main/kotlin/com/mudhut/nudge/users/repositories/PhoneVerificationTokenRepository.kt
package com.mudhut.nudge.users.repositories

import com.mudhut.nudge.users.entities.PhoneVerificationToken
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface PhoneVerificationTokenRepository : JpaRepository<PhoneVerificationToken, Long> {
    fun findByCode(code: String): Optional<PhoneVerificationToken>
}
